File: Sample shurjoPay SDK Test Java/app/src/main/java/com/example/testprojectapplication/MainActivity.java
package com.example.testprojectapplication;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.widget.Toast;

import com.example.testprojectapplication.databinding.ActivityMainBinding;
import com.shurjopay.sdk.v2.model.ErrorSuccess;
import com.shurjopay.sdk.v2.model.RequiredData;
import com.shurjopay.sdk.v2.payment.PaymentResultListener;
import com.shurjopay.sdk.v2.payment.ShurjoPaySDK;
import com.shurjopay.sdk.v2.utils.Constants;

import java.net.ContentHandler;
import java.util.Random;

public class MainActivity extends AppCompatActivity {
    private Activity activity;
    private Context context;
    private ActivityMainBinding binding;
    //
    private RequiredData requiredData;
    //

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        //
        binding.sysBtnPayment.setOnClickListener(view -> {
            makePayment();
        });
    }

    private void makePayment() {
        requiredData = getDataDefault();
        //
        ShurjoPaySDK.Companion.getInstance().makePayment(
                this,
                Constants.SDK_TYPE_SANDBOX,
                requiredData,
                new PaymentResultListener() {
                    @Override
                    public void onSuccess(@NonNull ErrorSuccess errorSuccess) {
                        Toast.makeText(
                                context,
                                "onSuccess: transactionInfo = " + errorSuccess.getTransactionInfo(),
                                Toast.LENGTH_LONG
                        ).show();
                    }

                    @Override
                    public void onFailed(@NonNull ErrorSuccess errorSuccess) {
                        Toast.makeText(
                                context,
                                "onFailed: transactionInfo = " + errorSuccess.getMessage(),
                                Toast.LENGTH_LONG
                        ).show();
                    }

                    @Override
                    public boolean onBackButtonListener(@NonNull ErrorSuccess errorSuccess) {
                        return false;
                    }
                }
        );
    }

    private RequiredData getDataDefault() {
        return new RequiredData(
                "username",
                "password",
                "prefix",
                "currency",
                1.0,
                "orderId",
                0.0,
                null,
                "customer_name",
                "customer_phone",
                "customerEmail",
                "customerAddress",
                "customerCity",
                "customerState",
                "customerPostcode",
                "customerCountry",
                null,
                null,
                null,
                null,
                null,
                null,
                null
        );
    }
}

File: sdk-v2/app/src/main/java/com/shurjopay/sdk/MainActivity.kt
package com.shurjopay.sdk

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.shurjopay.sdk.databinding.ActivityMainBinding
import com.shurjopay.sdk.v2.model.ErrorSuccess
import com.shurjopay.sdk.v2.model.RequiredData
import com.shurjopay.sdk.v2.payment.PaymentResultListener
import com.shurjopay.sdk.v2.payment.ShurjoPaySDK
import com.shurjopay.sdk.v2.utils.Constants
import java.util.*

class MainActivity : AppCompatActivity() {
    private val TAG = "MainActivity"
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.submitButton.setOnClickListener { pay() }
    }

    private fun pay() {
        val data = RequiredData(
            "sp_sandbox",
            "pyyk97hu&6u6",
            "NOK",
            "BDT",
            binding.amountLayout.editText?.text.toString().toDouble(),
            "NOK" + Random().nextInt(1000000),
            null,
            null,
            binding.nameLayout.editText?.text.toString(),
            binding.phoneLayout.editText?.text.toString(),
            null,
            binding.addressLayout.editText?.text.toString(),
            binding.cityLayout.editText?.text.toString(),
            null,
            null,
            null,
            "https://www.sandbox.shurjopayment.com/response",
            "https://www.sandbox.shurjopayment.com/response",
            "127.0.0.1",
            "value-of-1",
            "value-of-2",
            "value-of-3",
            "value-of-4"
        )

        ShurjoPaySDK.instance?.makePayment(
            this,
            Constants.SDK_TYPE_SANDBOX,
            data,
            object : PaymentResultListener {
                override fun onSuccess(errorSuccess: ErrorSuccess) {
                    Log.d(TAG, "onSuccess: debugMessage = ${errorSuccess.debugMessage}")
                }

                override fun onFailed(errorSuccess: ErrorSuccess) {
                    Log.d(TAG, "onFailed: debugMessage = ${errorSuccess.debugMessage}")
                }

                override fun onBackButtonListener(errorSuccess: ErrorSuccess): Boolean {
                    return true
                }
            })
    }
}
